### DISCUSSION TITLE
Authentication failure not able to push branch to fork repo

### DISCUSSION THREAD

**Discussion by Biswajit1693** [community/community#51721] [Repositories]
### Select Topic Area

Question

### Body

I forked a project repository and cloned it into my local machine using git clone.Earlier using my own repository the branch was main and when I cloned and entered the cloned repository the branch changed to master

I Edited the Readme.md file using vim inside the cloned directory

Created a new branch name project, checked out to the new branch and added the modified Readme.md to stage and then committed it.

I have added the remote upstream url using git remote add. But when trying to push it shows authentication failure. Not able to create or compare pull request.

Here are the steps I did:

1.git branch project

2.git checkout project

3.git add .

4.git commit -m "Changes done inside Readme.md"

In git logs its shows Head in project

When trying to push it - git push origin project. error-authentication failure

Kindly explain what I am doing wrong here?

I have added the classic token inside .git/config file of main branch
I can push my local repo to remote GitHub using the authentication and token for password section inside the .git/config file and can create commits inside my local repo which reflects in the remote GitHub page. But when I try to fork a public repo and clone it and edit and then push it using new branch then the error displays.

**Comment by tuves** [comment]
Hi @Biswajit1693,

The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your duplicative posts were removed as they violated our Community Code of Conduct, which includes posting the same comment in multiple places. As a result, we have deleted the duplicates, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).

**Comment by airtower-luna** [comment]
> But when I try to fork a public repo and clone it and edit and then push it using new branch then the error displays.

Where exactly are you trying to push to? Specifically, your fork or the upstream repository?
